### ISSUE TITLE
PROBLEM ABOUT THE DEPLOYMENT OF THE MODEL TO THE HARDWARE.

### ISSUE THREAD

**Issue by RonahJay-Emperio** [ultralytics/hub#603]
### Search before asking

- [X] I have searched the HUB [issues](https://github.com/ultralytics/hub/issues) and found no similar bug report.

### HUB Component

Models

### Bug

Hi! I tried to deploy the model to the hardware and it seems like the hardware cannot detect the classes based on the model that I created. I tried to test the model in the Ultralytics app and it works perfectly. My main concern is, how can we fix this issue and what are the possible cause of this error? I hope to get a fast response from you about this matter. Thank You!

### Environment

1. I use the iOS application for testing the models.
2. I used tensorFlow lite, ONNX, OpenVINO for deployment. But, none of these conversions worked.

**Comment by kalenmike** [comment]
@RonahJay-Emperio Thanks for raising the question. We are not able to help as we cannot see your code. We are currently working on getting the Ultralytics App code ready for open source, in the mean time we have a similair app already open sourced for iOS. My advice would be to take a look at the repo and see if there are any differences to how we load and run the model:

https://github.com/ultralytics/yolo-ios-app

**Comment by RonahJay-Emperio** [comment]
Hi! Thank You for your response. I used the Luxonis OAK FFC 4P device and it seems that the detection is not accurate because there's a lot of bounding boxes that are being detected during deployment. How can I detect the classes one at a time?

Here's the process I made before deployment:
1. Convert ONNX model into blob file.
2. Deploy the converted model into the device which is the OAK FFC 4P

Here's the sample code that I used:

from pathlib import Path
import cv2
import depthai as dai
import numpy as np
import time
import argparse

nnPathDefault = str((Path(__file__).parent / Path('../models/worker_activity_openvino_2022.1_6shave.blob')).resolve().absolute())
parser = argparse.ArgumentParser()
parser.add_argument('nnPath', nargs='?', help="Path to mobilenet detection network blob", default=nnPathDefault)
parser.add_argument('-s', '--sync', action="store_true", help="Sync RGB output with NN output", default=False)
args = parser.parse_args()

if not Path(nnPathDefault).exists():
    import sys
    raise FileNotFoundError(f'Required file/s not found, please run "{sys.executable} install_requirements.py"')

# MobilenetSSD label texts
labelMap = ["Walking", "At-Desk-Working", "Fall", "Running", "Sleeping", "Standing-NotWorking", "Standing-Working", "At-Desk-NotWorking"]
# Create pipeline
pipeline = dai.Pipeline()

# Define sources and outputs
camRgb = pipeline.create(dai.node.ColorCamera)
nn = pipeline.create(dai.node.NeuralNetwork)
det = pipeline.create(dai.node.DetectionParser)
xoutRgb = pipeline.create(dai.node.XLinkOut)
nnOut = pipeline.create(dai.node.XLinkOut)

xoutRgb.setStreamName("rgb")
nnOut.setStreamName("nn")

# Properties
camRgb.setPreviewSize(320, 320)
camRgb.setInterleaved(False)
camRgb.setFps(40)
# Define a neural network that will make predictions based on the source frames
nn.setNumInferenceThreads(2)
nn.input.setBlocking(False)

blob = dai.OpenVINO.Blob(args.nnPath)
nn.setBlob(blob)
det.setBlob(blob)
det.setNNFamily(dai.DetectionNetworkType.MOBILENET)
det.setConfidenceThreshold(1.0)

# Linking
if args.sync:
    nn.passthrough.link(xoutRgb.input)
else:
    camRgb.preview.link(xoutRgb.input)

camRgb.preview.link(nn.input)
nn.out.link(det.input)
det.out.link(nnOut.input)

# Connect to device and start pipeline
with dai.Device(pipeline) as device:
    # Output queues will be used to get the rgb frames and nn data from the outputs defined above
    qRgb = device.getOutputQueue(name="rgb", maxSize=4, blocking=False)
    qDet = device.getOutputQueue(name="nn", maxSize=4, blocking=False)

    frame = None
    detections = []
    startTime = time.monotonic()
    counter = 0
    color2 = (255, 255, 255)


    # nn data (bounding box locations) are in <0..1> range - they need to be normalized with frame width/height
    def frameNorm(frame, bbox):
        normVals = np.full(len(bbox), frame.shape[0])
        normVals[::2] = frame.shape[1]
        return (np.clip(np.array(bbox), 0, 1) * normVals).astype(int)


    def displayFrame(name, frame):
        color = (255, 0, 0)
        for detection in detections:
            if detection.label in range(7):
                bbox = frameNorm(frame, (detection.xmin, detection.ymin, detection.xmax, detection.ymax))
                cv2.putText(frame, labelMap[detection.label], (bbox[0] + 10, bbox[1] + 20), cv2.FONT_HERSHEY_TRIPLEX,0.5, color)
                # cv2.putText(frame, str(detection.label), (bbox[0] + 10, bbox[1] + 20), cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
                cv2.putText(frame, f"{int(detection.confidence * 100)}%", (bbox[0] + 10, bbox[1] + 40),cv2.FONT_HERSHEY_TRIPLEX, 0.5, color)
                cv2.rectangle(frame, (bbox[0], bbox[1]), (bbox[2], bbox[3]), color, 2)
        # Show the frame
        cv2.imshow(name, frame)


    while True:
        if args.sync:
            # Use blocking get() call to catch frame and inference result synced
            inRgb = qRgb.get()
            inDet = qDet.get()
        else:
            # Instead of get (blocking), we use tryGet (non-blocking) which will return the available data or None otherwise
            inRgb = qRgb.tryGet()
            inDet = qDet.tryGet()

        if inRgb is not None:
            frame = inRgb.getCvFrame()
            cv2.putText(frame, "NN fps: {:.2f}".format(counter / (time.monotonic() - startTime)),
                        (2, frame.shape[0] - 4), cv2.FONT_HERSHEY_TRIPLEX, 0.4, color2)

        if inDet is not None:
            detections = inDet.detections
            counter += 1

        # If the frame is available, draw bounding boxes on it and show the frame
        if frame is not None:
            displayFrame("rgb", frame)

        if cv2.waitKey(1) == ord('q'):
            break
            



![model_OverlapBoxes](https://github.com/ultralytics/hub/assets/113092301/596c4789-4f54-446d-8fb2-28553d3e1f60)
